### ISSUE TITLE
Ranges not working in proxy.config.http.connect_ports

### ISSUE THREAD

**Issue by mlibbey** [apache/trafficserver#11110]
https://docs.trafficserver.apache.org/admin-guide/files/records.config.en.html#proxy-config-http-connect-ports
says that proxy.config.http.connect_ports "Supports both wildcards (*) and ranges (e.g. 0-1023)."

However, ranges don't appear to be accepted:
```
[13:24:05  ~]$ sudo traffic_ctl config set proxy.config.http.connect_ports "443 4441-4451 8443"
Traffic Server: failed to set proxy.config.http.connect_ports: [14] Generic Fail message (ie. CoreAPI call).

[13:32:09  ~]$ sudo traffic_ctl config set proxy.config.http.connect_ports "443-4451"
Traffic Server: failed to set proxy.config.http.connect_ports: [14] Generic Fail message (ie. CoreAPI call).


[13:31:41  ~]$ sudo traffic_ctl config set proxy.config.http.connect_ports "443 8443 4441"
set proxy.config.http.connect_ports, please wait 10 seconds for traffic server to sync configuration, restart is not required
```

When adding to records.config directly, ATS doesn't complain -- so may only be a traffic_ctl issue.

**Comment by bryancall** [comment]
@mlibbey Just checking, is this for ATS 9.x?

**Comment by masaori335** [comment]
Above looks error with 9.2.x, but I also got below error with the master branch.
```
/opt/ats-asf-master-debug/bin/traffic_ctl config set proxy.config.http.connect_ports "443-4451"
Server Error found:
[9] Error during execution
- [2004] Validity check failed.
```

**Comment by masaori335** [comment]
It looks like the validation regex for the `proxy.config.http.connect_ports` doesn't allow `-`. This will be fixed by the #11149.

However, this means the validation is not working when the records.config is read. It's another issue needs to be fixed.
